### PR TITLE
Deprecate passing `PKey` instances to `use_privatekey`

### PR DESCRIPTION
**Pull request by alex**
Allow passing cryptography keys instead.

Refs #1321

### CODE REVIEW

File: src/OpenSSL/crypto.py [hunk 1]
```diff
@@ -84,6 +84,14 @@
...
 
 
+_PrivateKey = Union[
+    dsa.DSAPrivateKey,
+    ec.EllipticCurvePrivateKey,
+    ed25519.Ed25519PrivateKey,
+    ed448.Ed448PrivateKey,
+    rsa.RSAPrivateKey,
+]
+# TODO: Turn this into `PublicKey | PrivateKey` once we use `|`.
```
Comment (mhils) [inline src/OpenSSL/crypto.py:94]: Can't we do that right away?
```python3
>>> a = Union[1,2]
>>> b = Union[3,4]
>>> Union[a,b]
typing.Union[1, 2, 3, 4]
```

**Comment by alex** [comment]
Not on older pythons

**Comment by mhils** [comment]
```python
> docker run --rm -it python:3.7.0
Python 3.7.0 (default, Oct 16 2018, 07:10:55)
[GCC 6.3.0 20170516] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from typing import Union
>>> a = Union[str, int]
>>> b = Union[bool, float]
>>> Union[a,b]
typing.Union[str, int, bool, float]
```

This looks ok to me? (I've also tested 3.8-3.10)

**Comment by alex** [comment]
Ohhh, yeah, i guess I didn't try that. Will fix this evening

**Review by mhils — APPROVED** [review]

**Pull request merged**